### ISSUE TITLE
Popup triggers a regular login

### ISSUE THREAD

**Issue by jaxoncreed** [inrupt/solid-client-authn-js#424]
**Describe the bug**
When a config with `{ popUp: true }` is provided, the popup flow doesn't trigger. It will trigger the regular redirect login flow rather than making a popup window.

**To Reproduce**

1. Run the example at https://github.com/inrupt/solid-client-authn-js/tree/master/packages/browser/examples/single/bundle
2. Replace src/App.js with the code I wrote here: https://gist.github.com/jaxoncreed/f7554f4773886554abe719075df45fb6
3. Try to log in using the "Popup Login" button.

**Expected behavior**
A popup window should appear and should be directed to the issuer's login page. Upon login, the window closes and the session triggers an `onLogin` event.

**Environment**

```
System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 1.27 GB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.4.0 - /usr/local/bin/node
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 86.0.4240.80
    Firefox: 81.0
    Safari: 14.0
  npmPackages:
    @babel/core: ^7.8.6 => 7.10.5 
    @babel/preset-env: ^7.8.6 => 7.10.4 
    @babel/preset-react: ^7.8.3 => 7.10.4 
    @inrupt/solid-client-authn-browser: file://../../../ => 0.2.1 
    babel-loader: ^8.0.6 => 8.1.0 
    html-loader: ^0.5.5 => 0.5.5 
    html-webpack-plugin: ^3.2.0 => 3.2.0 
    react: ^16.13.0 => 16.13.1 
    react-dom: ^16.13.0 => 16.13.1 
    regenerator-runtime: ^0.13.3 => 0.13.7 
    webpack: ^4.41.6 => 4.44.0 
    webpack-cli: ^3.3.11 => 3.3.12 
    webpack-dev-server: ^3.10.3 => 3.11.0 
  npmGlobalPackages:
    @inrupt/generator-solid-react: 0.7.2
    @inrupt/solid-auth-fetcher: 0.0.6
    expo-cli: 3.27.4
    lerna: 3.22.1
    npm: 6.14.4
    ochat-api: 1.0.0
    redis-commander: 0.7.0
    static-server: 2.2.1
    typescript: 3.9.5
    yo: 3.1.1
```

**Additional context**
It looks like the functionality for popups was removed at some point. Probably because it wouldn't work very well with closures. I expect that this will be solved along with https://github.com/inrupt/solid-client-authn-js/issues/423. The popup window will trigger a login then close, causing the auth token to be lost but retaining the refresh token. Then the parent window would go through the refresh token flow to obtain an auth token for itself.

I need this for ESS compatibility. Unlike NSS, ESS will not accept hashes in its redirect URLs. So, if you log into https://jackson.inrupt.net/profile/card#me and want to get redirect there instead of https://jackson.inrupt.net/profile/card, ESS will complain. However, this same problem wouldn't happen in the Popup window flow.

**Comment by NSeydoux** [comment]
Indeed, the support for popup login has been currently removed, and we are considering our options to add it back. As you mention, the necessity to have inter-process communication between the popup and the browser window challenges the way the token is stored never to be exposed in the current implementation. I've seen some best practices documents frowning upon popup login in general, so we'll see what tradeoff can be made and if it is worth it.

**Comment by jaxoncreed** [comment]
Hmm. If popup is discouraged, I could work around that.

**Comment by jaxoncreed** [comment]
Though, for the data-browser, without popup login, if you have navigated into the folder structure without logging in and you want to then login, it's going to pop you back at the root of that page instead of redirecting you back to the thing you were looking at ☹️

**Comment by jaxoncreed** [comment]
Tim is saying that he'd like login to open in a new tab rather than a popup.

**Comment by Vinnl** [comment]
The way the data browser works is that it does actually display the resource you were looking at. However, you're probably using the folder pane, which allows you to drill down into the folder represented by the current URL - but the folder pane doesn't maintain its state across refreshes, nor does it represent it in the URL.

I'm afraid that whether it's in a new tab or in a popup doesn't matter in terms of avoiding the inter-process communication, so @NSeydoux's comment above would still apply.

**Comment by jaxoncreed** [comment]
Generally, when it comes to databrowser implementation it comes down to a choice between 3 things:

 - Provide some kind of way to log in without redirecting the current window in this library (it could be popup or a new tab). This would take extra work on the part of the solid-client-authn-js team.
 - Just accept that the databrowser will no longer maintain its state after you login. This would be inconvenient if you're browsing something nested in the folder pane or you're looking at something in the pane.
 - Before redirecting save the state of the application so it can be loaded back later. This would take a lot of work because the databrowser doesn't currently save its state anywhere and doing this would require a rearchitecture.

Let me know if I charactarized that correctly.

**Comment by Vinnl** [comment]
I'd add to the first bullet that it's not just extra work, but that we don't yet know how to do this securely. Your suggestion at https://github.com/inrupt/solid-client-authn-js/issues/423#issuecomment-709295748 looks promising and might, I think, be able to help in this regard, but we have yet to investigate whether that satisfies the security requirements. So fingers crossed :)

**Comment by NSeydoux** [comment]
Also, doing a full loggin in an `iframe` is technically impossible in a lot of cases, because logging endpoints such as Google's actively prevent that to protect against phishing.

I think your three points capture well the tradeoff @jaxoncreed, with the caveat brought by @Vinnl. Returning the token back to the main window from the popup would require some work, as it does not fit with the current way the token is handled for security reasons, so that would effectively mean that login to the data browser cannot be done securely. Combined with the fact that the modular architecture makes it possible to have scripts from all over the place running together in the page, that might be a big concern. However, I did not have the time yet to look into the alternatives you suggest @jaxoncreed, maybe they could resolve the issue. Otherwise, on a short term, the only option that looks achievable to me is not to maintain state, which makes a terrible user experience...

**Comment by josephguillaume** [comment]
While maintaining DOM state might not be possible, persisting state through the authentication process should be possible but isn't implemented yet (https://github.com/inrupt/solid-client-authn-js/issues/950).

At the moment it seems even that is not possible - URL parameters or hash links would not survive the login process? (and it is needed at every page refresh #423)
